enhancements/rhcos/static-networking-enhancements.md

```diff
@@ -0,0 +1,104 @@
+
+This is basic networking configuration and users assume basic configurations like this are possible. As RHCOS requires a functional networking config early in the boot process, this eliminates many of the existing mechanisms customers use to handle the assigning of static networking configs. Most of the OpenShift personas we target are not aware of `dracut` arguments and it’s not a great experience to expect them to understand very low-level details of how the OS works.
+
+### Goals
+  - Provide a guided configuration of networking and Ignition source information
+  - Provide tooling to create customized ISOs that include unique Ignition configurations
+  - Provide support for a VMWare "backchannel" to provide network configuration and Ignition source information
+
+### Non-Goals
+  - Providing a fully automated solution for providing static network configuration to a fleet of machines
+
+## Proposal
+For the bare metal use case, we are proposing the use of the new [live ISO](https://github.com/openshift/enhancements/pull/210) to provide an interactive environment where users can configure their networking parameters and have them persist to the real root.  In this environment, users will be able to invoke a TUI that will allow them to configure networking and Ignition source.  This information will be used as part of the install of the host and will persist into the real root.
+
+Additionally, the new live ISO will provide the ability to embed the networking configuration via an Ignition config, resulting in a custom ISO that can be used to install to a host in an automated fashion.
+
+For the VMware use case, the proposal is to the existing `guestinfo` method to provide the static networking configuration.
+
+### User Stories
+
+#### Story 1
+A user wants to spin up a single FCOS node on bare metal using a static networking configuration.  They boot the live ISO and are dropped into an interactive FCOS environment.  They are able to use a tool to guide them to configure static networking for the host and provide a URL to the Ignition configuration.  The user runs `coreos-installer` after the configuration is complete and the host boots into the newly installed OS with static networking configured and Ignition config applied.
+
+### Story 2
+A customer has a fleet of bare metal machines they want to provision with RHCOS and static networking, that will later be used for the installation of OpenShift Container Platform (OCP).  They use some automation to generate unique Ignition configs that include the static networking configuration.  Using similar automation, they use `coreos-installer iso embed` to embed the Ignition configs into unique ISOs per machine.  These ISOs can then be attached to the baseboard machine controller (BMC) or lights out management (LOM) of the host, at which point the hosts can be booted and RHCOS will be automatically installed with the static networking configured and Ignition config applied.
```

Comment:
I assume in this scenario the user is embedding an Ignition config that will do an automated install like what is described in https://dustymabe.com/2020/04/04/automating-a-custom-install-of-fedora-coreos/ ? I think there are corner cases here for networking in the initramfs that we need to account for in our documentation. I can elaborate on that here or in another forum if needed.

Reply:
I think those corner cases should be covered as part of the official docs; we don't need to get too specific as part of the enhancement.